### ISSUE TITLE
with_tag in a block is not restricted to the found element

### ISSUE THREAD

**Issue by randoum** [kucaahbe/rspec-html-matchers#58]
```
is_expected.to have_tag('div#menu') {
  with_tag 'a.hidden', count: 1
}
```
Do find all `a.hidden` in the entire document. 

With 0.8.2 only, does not append with 0.8.1
I suspect #55 to be responsible

**Comment by randoum** [comment]
ping @kucaahbe @misaka

**Comment by jcouball** [comment]
I believe this is a duplicate of issue #41

**Comment by jcouball** [comment]
:+1: I see this issue too.  I have had to pin to the 0.8.1 version of the gem as a workaround.

**Comment by randoum** [comment]
@jcouball would you mind to check #59 and see if your issues are fixed ?

**Comment by jcouball** [comment]
@randoum the changes in #59 seem to fix the problems with my tests.  Thank you!

**Comment by randoum** [comment]
Fixed by #59
